server/utils/file/csvReader.js

```diff
@@ -5,25 +5,38 @@ const Queue = require('../../../core/queue')
 
 const createReaderFromStream = (stream, onHeaders = null, onRow = null, onTotalChange = null) => {
 
-  let headers = null
-  let total = 0
   let canceled = false
   const queue = new Queue()
 
   const start = () => new Promise((resolve, reject) => {
     let started = false
     let ended = false
+    let headers = null
+    let total = 0
+    let processingRow = false //prevents the call to processNext when a row is already being processed
 
     const processNext = () => {
-      (async () => {
-        if (queue.isEmpty()) {
-          if (ended)
-            resolve()
-        } else if (!canceled) {
-          onRow && await onRow(queue.dequeue())
+      if (queue.isEmpty()) {
+        if (ended)
+          resolve()
+      } else if (!canceled) {
+        if (onRow) {
+          processingRow = true
+          onRow(queue.dequeue())
+            .then(() => {
+              processNext()
+            })
+            .catch(e => {
+              cancel()
+              reject(e)
+            })
+            .finally(() => {
+              processingRow = false
+            })
```

Comment:
can't this be used with async/await ? what was the problem here?

Reply:
in the end it doesn't change too much, I'll revert to the previous version and add the processingRow handling